features/base/participants/reducer.js

```diff
@@ -73,45 +73,27 @@ function participant(state, action) {
         }
         return state;
 
-    /**
-     * Cleans conference-specific properties of the local participant when
-     * conference is left.
-     */
-    case CONFERENCE_LEFT:
-        if (state.local) {
-            return {
-                ...state,
-                id: LOCAL_PARTICIPANT_DEFAULT_ID,
-                avatar: state.avatar ||
-                    _getAvatarURL(LOCAL_PARTICIPANT_DEFAULT_ID, state.email),
-                focused: false,
-                pinned: false,
-                role: PARTICIPANT_ROLE.NONE,
-                selected: false,
-                speaking: false
-            };
-        }
-        return state;
-
-    case DOMINANT_SPEAKER_CHANGED:
-        // Only one dominant speaker is allowed.
-        return Object.assign({}, state, {
-            speaking: state.id === action.participant.id
-        });
-
     case PARTICIPANT_JOINED:
+        let participant = action.participant;
+        let participantId = participant.id
+            || (participant.local && LOCAL_PARTICIPANT_DEFAULT_ID);
```

Comment:
Yes, the logic if following:
1) we provided a participantId
2) we didn't provide a participantId:
    a) it's a local participant, then we use LOCAL_PARTICIPANT_DEFAULT_ID
    b) if it's not a local participant, we leave participant ID undefined.

I can add this comment to code to make it more clear.